File: benchmark/corpus/python/secret_safe_1.py
"""SAFE: credentials loaded from environment variables, not hardcoded."""
import os
import requests


def fetch_data(endpoint: str):
    api_key = os.environ["API_KEY"]
    headers = {"Authorization": f"Bearer {api_key}"}
    response = requests.get(endpoint, headers=headers)
    return response.json()


def get_db_dsn() -> str:
    host = os.environ.get("DB_HOST", "localhost")
    user = os.environ.get("DB_USER", "app")
    password = os.environ["DB_PASSWORD"]
    return f"postgresql://{user}:{password}@{host}:5432/myapp"

File: benchmark/corpus/go/secret_safe_1.go
// SAFE: credentials read from environment variables at runtime
package main

import (
	"fmt"
	"net/http"
	"os"
)

func makeRequest(endpoint string) {
	apiKey := os.Getenv("API_KEY")
	if apiKey == "" {
		fmt.Fprintln(os.Stderr, "API_KEY not set")
		return
	}
	req, _ := http.NewRequest("GET", endpoint, nil)
	req.Header.Set("Authorization", "Bearer "+apiKey)
	http.DefaultClient.Do(req)
}
